crates/iroha_executor/src/default/custom/multisig/transaction.rs

```diff
@@ -0,0 +1,254 @@
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                instructions_key(&instructions_hash),
+            ))
+            .dbg_unwrap()
+            .try_into_any()
+            .dbg_unwrap();
+        let proposed_at_ms: u64 = host
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                proposed_at_ms_key(&instructions_hash),
+            ))
+            .dbg_unwrap()
+            .try_into_any()
+            .dbg_unwrap();
+        let mut approvals: BTreeSet<AccountId> = host
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                approvals_key(&instructions_hash),
+            ))
+            .dbg_unwrap()
+            .try_into_any()
+            .dbg_unwrap();
+
+        approvals.insert(approver);
+
+        host.submit(&SetKeyValue::account(
+            target_account.clone(),
+            approvals_key(&instructions_hash),
+            Json::new(&approvals),
+        ))
+        .dbg_unwrap();
+
+        let now_ms: u64 = executor
+            .context()
+            .curr_block
+            .creation_time()
+            .as_millis()
+            .try_into()
+            .dbg_expect("shouldn't overflow within 584942417 years");
+
+        let is_authenticated = quorum
+            <= signatories
+                .into_iter()
+                .filter(|(id, _)| approvals.contains(id))
+                .map(|(_, weight)| u16::from(weight))
+                .sum();
+
+        let is_expired = proposed_at_ms.saturating_add(transaction_ttl_ms) < now_ms;
+
+        if is_authenticated || is_expired {
+            // Cleanup the transaction entry
+            host.submit(&RemoveKeyValue::account(
+                target_account.clone(),
+                approvals_key(&instructions_hash),
+            ))
+            .dbg_unwrap();
+
+            host.submit(&RemoveKeyValue::account(
+                target_account.clone(),
+                proposed_at_ms_key(&instructions_hash),
+            ))
+            .dbg_unwrap();
+
+            host.submit(&RemoveKeyValue::account(
+                target_account.clone(),
+                instructions_key(&instructions_hash),
+            ))
+            .dbg_unwrap();
+
+            if is_expired {
+                // TODO Notify that the proposal has expired, while returning Ok for the entry deletion to take effect
+            } else {
+                // Validate and execute the authenticated multisig transaction
+                for instruction in instructions {
+                    let init_authority = executor.context().authority.clone();
+                    // Authorize as the multisig account
+                    executor.context_mut().authority = target_account.clone();
+                    executor.visit_instruction(&instruction);
+                    executor.context_mut().authority = init_authority;
+                }
```

Comment:
```suggestion
                for instruction in instructions {
                    let mut executor = executor.clone();
                    // Authorize as the multisig account
                    executor.context_mut().authority = target_account.clone();
                    executor.visit_instruction(&instruction);
                    executor.verdict()?
                }
```

I actually wanted to do this to make executor immutable during `execute`, but it would introduce an extra `Clone` trait bound:

```rust
    fn execute<V: Clone + Execute + Visit + ?Sized>(self, executor: &V) -> Result<(), ValidationFail>;
```